tcping.go

```diff
@@ -632,9 +639,36 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, connTime time.Time) {
 	)
 }
 
+func getInterfaceAddr(tcpStats *stats) (addr string) {
+	var ipAddr net.IP
+	ief, err := net.InterfaceByName(tcpStats.userInput.interfaceName)
+	if err != nil {
+		tcpStats.printer.printError("Interface name %s not allowed", tcpStats.userInput.interfaceName)
+	}
+	addrs, err := ief.Addrs()
+	if err != nil {
+		tcpStats.printer.printError("Unable to get Interface address")
+	}
+	for _, addr := range addrs {
+		if ipAddr = addr.(*net.IPNet).IP; ipAddr != nil {
+			break
+		}
+	}
+	if ipAddr == nil {
+		tcpStats.printer.printError("Unable to generate Interface's IP Address")
+	}
+	return ipAddr.String()
+}
+
```

Comment:
I would also move this way up the code base, ideally below the `processUserInput` fucntion